src/customRules/importTypeOrderRule.ts

```diff
@@ -0,0 +1,75 @@
+import * as ts from 'typescript';
+import * as Lint from 'tslint';
+
+export class Rule extends Lint.Rules.AbstractRule {
+  /* tslint:disable:object-literal-sort-keys */
+  public static metadata: Lint.IRuleMetadata = {
+    'ruleName': 'import-type-order',
+    'description': 'Enforce structure to your imports. Import structure should be listed in the following order: modules, absolute imports,  relative parent/ancestor directories, relative sibling directors.',
+    'hasFix': false,
+    'optionsDescription': 'Not configurable.',
+    'options': null,
+    'optionExamples': null,
+    'type': 'style',
+    'typescriptOnly': true,
+  };
+  /* tslint:enable:object-literal-sort-keys */
+
+  public static IMPORT_TYPE_ORDER_ERROR = 'Imports should be listed in the following order: module imports, absolute imports, ancestor imports, sibling imports.';
+
+  public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+    const structuredImportsWalker = new StructuredImportsWalker(sourceFile, this.ruleName, undefined);
+    return this.applyWithWalker(structuredImportsWalker);
+  }
+}
+
+enum ImportType {
+  External = 0, // 'external'
+  Absolute = 1, // '/some/folder/file'
+  Ancestor = 2, // '../parentFolder'
+  Sibling = 3, // './siblingFolder'
+};
+
+const importStuctureOrder = [ImportType.External, ImportType.Absolute, ImportType.Ancestor, ImportType.Sibling];
+
+class StructuredImportsWalker extends Lint.AbstractWalker<void>  {
+  private previousImport: ImportType;
```

Comment:
Haha, right!  Happy states are good practice.  Sorry, too quick on the trigger 😛